src/RecursiveIteratorLocator.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace RebelCode\Modular\Locator;
+
+use Dhii\I18n\StringTranslatorAwareTrait;
+use Dhii\I18n\StringTranslatorConsumingTrait;
+use Dhii\I18n\StringTranslatorInterface;
+use Dhii\Modular\Locator\ModuleLocatorExceptionInterface;
+use Dhii\Modular\Locator\ModuleLocatorInterface;
+use Dhii\Validation\Exception\ValidationFailedExceptionInterface;
+use Dhii\Validation\ValidatorInterface;
+use Exception;
+use RebelCode\Modular\Config\ConfigInterface as Cfg;
+use RecursiveIterator;
+
+/**
+ * A module locator implementation that retrieves module configuration data from a recursive iterator or an iterator
+ * of iterators.
+ *
+ * @since [*next-version*]
+ */
+class RecursiveIteratorLocator extends AbstractRecursiveIteratorLocator implements ModuleLocatorInterface
+{
+    /*
+     * @since [*next-version*]
+     */
+    use StringTranslatorAwareTrait;
+
+    /*
+     * @since [*next-version*]
+     */
+    use StringTranslatorConsumingTrait;
+
+    /**
+     * The validator used to validate configuration.
+     *
+     * @since [*next-version*]
+     *
+     * @var ValidatorInterface
+     */
+    protected $configValidator;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param RecursiveIterator         $iterator   The recursive iterable containing the module config iterators.
+     * @param ValidatorInterface        $validator  The validator to use for config validation.
+     * @param StringTranslatorInterface $translator The translator to use for validation messages and logging.
+     */
+    public function __construct(
+        RecursiveIterator $iterator,
```

Comment:
Didn't quite understand what it iterates over. The iterator doesn't even seem to be getting set anywhere.